### PR TITLE
sql/pgwire: send CommandComplete for DDLs

### LINKED ISSUE #3849: pgwire: empty query vs command ok
Looks like we're sending `PGRES_EMPTY_QUERY` instead of `PGRES_COMMAND_OK` for a `drop table if exists...`, causing code that checked for `PGRES_COMMAND_OK`, eg the setup in `pgbench`, to fail.

### PR DESCRIPTION
**Pull request by dt**
Fixes #3849

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3853)

### CODE REVIEW

File: sql/pgwire/v3.go [hunk 1]
```diff
@@ -677,6 +672,7 @@ func (c *v3Conn) sendResponse(resp driver.Response, formatCodes []formatCode, se
 
 		// Ack messages do not have a corresponding protobuf field, so handle those
 		// with a default.
+		// This also includes DDLs which want CommandComcplete as well.
```
Comment (tamird) [inline sql/pgwire/v3.go]: typo

**Comment by tamird** [comment]
Sigh, CI fails because `serverMsgEmptyQuery` is unused now. Perhaps you could fix #3852 while you're here.

**Comment by dt** [comment]
As discussed on #3852, a "real" fix three is going go involve plumbing the parser's opinion on emptiness though sql/driver. Once that's done the naive check added here might go, or stay around as it avoid work in the simple case.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---

*[sql/pgwire/v3.go, line 675 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K80Tows4Fm1eGi2daLf-r1-675:-K86UWdKkLWs1npipI5V:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/ee8606712441677c3f11efe01164a0605ebbe3bc/sql/pgwire/v3.go#L675)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-:-K86UZlB-_pA1TIti4h0:-1277479441)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[sql/pgwire/v3.go, line 548 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K86lpWWSM4bT94ocCch:-K86lpWX07mjHMkHd6MI:-35193107) ([raw file](https://github.com/cockroachdb/cockroach/blob/7e0f5f410aa339e696714b63ecc5a227cf6c4e07/sql/pgwire/v3.go#L548)):*
nit: don't you usually use `dt`? also, for greppability, no space between TODO and the paren

---

*[sql/pgwire/v3.go, line 550 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K86lwq63TcpNtg5DRgD:-K86lwq63TcpNtg5DRgE:-938752909) ([raw file](https://github.com/cockroachdb/cockroach/blob/7e0f5f410aa339e696714b63ecc5a227cf6c4e07/sql/pgwire/v3.go#L550)):*
unless execute?

---

*[sql/pgwire/v3.go, line 680 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K86ljgjgAHwglYYUuTO:-K86ljgjgAHwglYYUuTP:-1967442619) ([raw file](https://github.com/cockroachdb/cockroach/blob/7e0f5f410aa339e696714b63ecc5a227cf6c4e07/sql/pgwire/v3.go#L680)):*
put your name on this TODO

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Comment by dt** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/pgwire/v3.go, line 548 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K86lpWWSM4bT94ocCch:-K86mCJGSH0rIqkkLR-g:-1426617367) ([raw file](https://github.com/cockroachdb/cockroach/blob/7e0f5f410aa339e696714b63ecc5a227cf6c4e07/sql/pgwire/v3.go#L548)):*
oh, yeah, old habits. Done.

---

*[sql/pgwire/v3.go, line 550 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K86lwq63TcpNtg5DRgD:-K86mwoAgBm282QJ2SPV:21823127) ([raw file](https://github.com/cockroachdb/cockroach/blob/7e0f5f410aa339e696714b63ecc5a227cf6c4e07/sql/pgwire/v3.go#L550)):*
no-op. clarified.

---

*[sql/pgwire/v3.go, line 680 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K86ljgjgAHwglYYUuTO:-K86mkJdIvYeR_DxOr_s:1716178427) ([raw file](https://github.com/cockroachdb/cockroach/blob/7e0f5f410aa339e696714b63ecc5a227cf6c4e07/sql/pgwire/v3.go#L680)):*
bah, no good deed goes unpunished.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Comment by tamird** [comment]
This needs a test, if that's feasible.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-:-K86wShej9LBJROlHx6R:-1223401348)*

**Comment by dt** [comment]
I'll take another look at`lib/pq` but it might not be easily within our existing tests.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-:-K86wvge4XEZXVV-3h8K:-694232487)*

**Comment by dt** [comment]
After digging through lib/pq and database/sql, figured out a test but it is *gross*.

---

Review status: 1 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-:-K8WswsduHdGU76MNGK8:-1909614602)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[sql/pgwire_test.go, line 333 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K8ZJKYTGvpeiWzVxlpf:-K8ZJKYTGvpeiWzVxlpg:1726150812) ([raw file](https://github.com/cockroachdb/cockroach/blob/1db939e3cd416697d3f1cd1fb274df378ba8b7fa/sql/pgwire_test.go#L333)):*
s/PGWire/CmdCompleteVsEmptyStatements/

---

*[sql/pgwire_test.go, line 346 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K8ZJ3qojnqYCmD1-Jze:-K8ZJ3qojnqYCmD1-Jzf:1142582758) ([raw file](https://github.com/cockroachdb/cockroach/blob/1db939e3cd416697d3f1cd1fb274df378ba8b7fa/sql/pgwire_test.go#L346)):*
there needs to be a comment here explaning what is actually being tested. in other words, the reader needs to know that schema changes should not return empty query responses.

also, why is creating the table necessary? seems like the `CREATE DATABASE` above would suffice.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Comment by dt** [comment]
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/pgwire_test.go, line 333 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K8ZJKYTGvpeiWzVxlpf:-K8_1URCReaA13NRYAVq:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/1db939e3cd416697d3f1cd1fb274df378ba8b7fa/sql/pgwire_test.go#L333)):*
Done.

---

*[sql/pgwire_test.go, line 346 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K8ZJ3qojnqYCmD1-Jze:-K8_1Vr_9wvkTitEFEdZ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/1db939e3cd416697d3f1cd1fb274df378ba8b7fa/sql/pgwire_test.go#L346)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/pgwire_test.go, line 365 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K8dasC20MdXptKBiWNT:-K8dasC3_vd3MbGdCz30:-1514943174) ([raw file](https://github.com/cockroachdb/cockroach/blob/93d48cd7f552f091abc38e1b840be3152c627444/sql/pgwire_test.go#L365)):*
if you comment this line out - does the test fail?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Comment by dt** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/pgwire_test.go, line 365 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K8dasC20MdXptKBiWNT:-K8eaDxVhtrNoIrEbIk6:677251183) ([raw file](https://github.com/cockroachdb/cockroach/blob/93d48cd7f552f091abc38e1b840be3152c627444/sql/pgwire_test.go#L365)):*
Yep.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[sql/pgwire_test.go, line 365 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853#-K8dasC20MdXptKBiWNT:-K8eb_K7VnatUl37Qe6r:-1264548050) ([raw file](https://github.com/cockroachdb/cockroach/blob/93d48cd7f552f091abc38e1b840be3152c627444/sql/pgwire_test.go#L365)):*
Really? I'm surprised the `defer`ed `recover()` doesn't swallow the `Fatal` below. LGTM then.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3853)*

**Pull request merged**